misc/ballerina-bindgen/src/main/java/org/ballerinalang/bindgen/command/BindingsGenerator.java

```diff
@@ -140,17 +139,30 @@ void generateJavaBindings() throws BindgenException {
     }
 
     private void resolvePlatformLibraries() throws BindgenException {
-        if (projectRoot != null) {
-            Manifest manifest = TomlParserUtils.getManifest(projectRoot);
-            if (manifest != null) {
-                List<Library> platformLibraries = manifest.getPlatform().getLibraries();
-                if (platformLibraries != null) {
-                    for (Library library : platformLibraries) {
-                        if (library.path != null) {
-                            classPaths.add(Paths.get(projectRoot.toString(), library.path).toString());
-                        } else if (library.groupId != null && library.artifactId != null && library.version != null) {
-                            new BindgenMvnResolver(outStream).mavenResolver(library.groupId, library.artifactId,
-                                    library.version, projectRoot, false);
+        if (env.getProjectRoot() != null) {
+            TomlDocument tomlDocument = env.getTomlDocument();
+            if (tomlDocument != null) {
+                PackageManifest packageManifest = ManifestBuilder.from(tomlDocument, null,
```

Comment:
Shall we refactor the nested if and for we can split it in to multiple functions

Reply:
Done